samples/wavlib/src/main/cpp/player/OneShotSampleBuffer.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "io/wav/WavStreamReader.h"
+
+#include "OneShotSampleBuffer.h"
+
+namespace wavlib {
+
+void OneShotSampleBuffer::loadSampleData(WavStreamReader* reader) {
+    mProperties.channelCount = reader->getNumChannels();
+    mProperties.sampleRate = reader->getSampleRate();
+
+    reader->positionToAudio();
+
+    // For now we know that the source sample format is 16bit
```

Comment:
Why 16 bit?

Reply:
Because all the samples you provided are 16-bit and 44.1K. You are right thought that this knowledge doesn't belong here. Let me sort it out.

Reply:
Turns out the assumption is irrelevant since we use getDataFloat() to retrieve the data from the SampleBuffer.